### PR TITLE
[Component with its own class name as service type] LPS-194841 Refactor Interpreter component

### PR DESCRIPTION
**Pull request by Valmir-unicap**
Related: https://liferay.atlassian.net/browse/LPS-194841

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:search

**Comment by Valmir-unicap** [comment]
ci:test:lima

File: modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/SharingEntryInterpreterProviderImpl.java [hunk 1]
```diff
@@ -70,13 +91,106 @@ private boolean _isAssetObject(String className) {
...
+	private static final Log _log = LogFactoryUtil.getLog(
+		SharingEntryInterpreterProviderImpl.class);
+
 	@Reference
-	private AssetRendererSharingEntryInterpreter
-		_assetRendererSharingEntryInterpreter;
+	private AssetEntryLocalService _assetEntryLocalService;
+
+	private AssetRendererSharingEntryEditRenderer
+		_assetRendererSharingEntryEditRenderer;
```
Comment (rafaprax) [inline modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/SharingEntryInterpreterProviderImpl.java:101]: @Valmir-unicap move renderers to the inner class

File: modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/SharingEntryInterpreterProviderImpl.java [hunk 2]
```diff
@@ -70,13 +91,106 @@ private boolean _isAssetObject(String className) {
...
+	private final AssetRendererSharingEntryInterpreter
+		_assetRendererSharingEntryInterpreter =
+			new AssetRendererSharingEntryInterpreter();
```
Comment (rafaprax) [inline modules/apps/sharing/sharing-web/src/main/java/com/liferay/sharing/web/internal/interpreter/SharingEntryInterpreterProviderImpl.java:104]: initialize this field into the activate method.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Pull request closed without merging**